R/input-slider.R

```diff
@@ -227,6 +282,164 @@ hasDecimals <- function(value) {
 }
 
 
+accentVariableDefaults <- function(skin, accentColor = NULL) {
+  if (useBsTheme()) {
+    primary <- if ("3" %in% bootstraplib::theme_version()) "$brand-primary" else "$primary"
+    accentColor <- parseAccentColor(accentColor %OR% primary)
+    accentColorDark <- paste0("mix(", accentColor, ", $black, 80%)")
+    accentColorLight <- paste0("mix(", accentColor, ", $white, 10%)")
+  } else {
+    accentColor <- parseAccentColor(accentColor %OR% defaultAccentColor(skin))
+    accentColorDark <- paste0("mix(", accentColor, ", black, 80%)")
+    accentColorLight <- paste0("mix(", accentColor, ", white, 10%)")
+  }
+  vars <- switch(skin,
+    big = list(
+      "bar_color"     = accentColor,
+      "label_color_1" = accentColor,
+      "grid_color_1"  = accentColor
+    ),
+    flat = list(
+      "bar_color"      = accentColor,
+      "handle_color_1" = accentColorDark,
+      "handle_color_2" = accentColorDark,
+      "label_color_1"  = accentColor
+    ),
+    modern = list(
+      "bar_color"     = accentColor,
+      "label_color_1" = accentColor
+    ),
+    round = list(
+      "bar_color"      = accentColor,
+      "handle_color_1" = accentColor,
+      "handle_color_3" = accentColorLight,
+      "label_color_1"  = accentColor
+    ),
+    sharp = list(
+      "bar_color"       = accentColor,
+      "handle_color_1"  = accentColorDark,
+      "minmax_bg_color" = accentColorDark,
+      "label_color_1"   = accentColorDark
+    ),
+    shiny = list("accent" = accentColor),
+    square = list(
+      "bar_color"      = accentColor,
+      "handle_color_1" = accentColor,
+      "handle_color_3" = accentColorLight,
+      "label_color_1"  = accentColor
+    ),
+    stop("Couldn't find skin named '", skin, "'", call. = FALSE)
+  )
+
+  lapply(vars, function(x) paste(x, "!default"))
+}
+
+# TODO: BS3 version!
+bsThemeSliderDefaults <- function(skin) {
+  if (!useBsTheme()) {
+    return("")
+  }
+
+  vars <- switch(skin,
+    big = list(
+      "line_color_1"      = "$white",
+      "line_color_2"      = "gray('200')",
+      "line_color_3"      = "gray('100')",
+      "handle_color_1"    = "gray('200')",
+      "handle_color_2"    = "gray('300')",
+      "handle_color_3"    = "$white",
+      "handle_color_4"    = "gray('400')",
+      "minmax_text_color" = "$white",
+      "minmax_bg_color"   = "gray('500')",
+      "label_color_2"     = "$white"
```

Comment:
Turns out that Bootswatch doesn't set `$white`/`$black` to true "bg"/"fg" colors, so I think we should replace `$white` with `$body-bg` and `$black` with `$body-color`